std/range/package.d

```diff
@@ -8606,6 +8606,302 @@ if (isForwardRange!Source && hasLength!Source)
     assert(equal(chunks, [[1], [2], [3], [], []]));
 }
 
+/**
+This range splits the source range into chunks of the sizes specified into the
+chunkSizes range. Both the source range and the sizes range must be input ranges.
+
+If the source range is shorter than the sum of the sizes, only the sizes that
+cover the range will be used. In case the last available elements of the source
+range don't reach the specified size, the chunk will only contain those elements.
+
+Params:
+    source = range from which the chunks will be selected
+    chunkSizes = range specifying the size of each chunk
+*/
+struct VarChunks(Source, Sizes)
+if (isInputRange!Source && isInputRange!Sizes && isIntegral!(ElementType!Sizes))
+{
+    static if (isForwardRange!Source && isForwardRange!Sizes)
+    {
+        this(Source source, Sizes chunkSizes)
+        {
+            _source = source;
+            _chunkSizes = chunkSizes;
+        }
+
+        // Input range primitives
+        static if (!isInfinite!(Source))
+            @property bool empty()
+            {
+                static if (!isInfinite!(Sizes))
+                    return _source.empty || _chunkSizes.empty;
```

Comment:
I was thinking that if not enough sizes are provided then the rest of the elements are not needed. But it might be a better idea to just make a chunk out of all the remaining elements so that those are not simply lost.

Reply:
That sounds like a more sensible approach and it also matches the existing `chunks` behaviour (which will return a smaller chunk if there are not enough remaining elements)